File: Data Structures & Algorithms/meeting-schedule-ii/submission-0.py
"""
Definition of Interval:
class Interval(object):
    def __init__(self, start, end):
        self.start = start
        self.end = end
"""

class Solution:
    def minMeetingRooms(self, intervals: List[Interval]) -> int:

        intervals.sort(key=lambda x:x.end)

        res = intervals

        count = 0

        while res:

            p_res = []
            prev_end = res[0].end
            for i in range(1,len(res)):

                if res[i].start<prev_end:
                    p_res.append(res[i])
                else:
                    prev_end = res[i].end

            res = p_res
            count+=1

        return count 
        

File: Data Structures & Algorithms/permutation-string/submission-0.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {

        int l1 = s1.length();
        int l2 = s2.length();

        vector<int> c1(26,0);
        vector<int> c2(26,0);

        for(auto s:s1){

            c1[s-'a']++;
        }

        for(int i=0;i<l2;i++){

            c2[s2[i]-'a']++;
            if(i>=l1){

                c2[s2[i-l1]-'a']--;
            }
            if(c1==c2){
                return true;
            }
        }

        return false;


        
    }
};
